src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleAuthScreen.java

```diff
@@ -53,7 +53,9 @@ public GoogleAuthScreen() {
 
   private ActionListener onSignin() {
     return e -> {
-      var authReqRes = handleGoogleSignIn();
+      var currentCredentials = handleGoogleSignIn();
+      var authReqRes =
+          currentCredentials.isTokenExpired() ? refreshGoogleCredentials() : handleGoogleSignIn();
```

Comment:
put this logic in the google Api because it looks weird to handle signin twice

Reply:
done